src/lib/fs-utils.ts

```diff
@@ -1,42 +1,90 @@
-import * as Bluebird from 'bluebird';
 import * as _ from 'lodash';
-import { fs } from 'mz';
-import * as Path from 'path';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+
 import * as constants from './constants';
-import { ENOENT } from './errors';
 
-export function writeAndSyncFile(
-	path: string,
+export const execAsync = promisify(exec);
+
+export async function writeAndSyncFile(
+	_path: string,
 	data: string | Buffer,
-): Bluebird<void> {
-	return Bluebird.resolve(fs.open(path, 'w')).then((fd) => {
-		_.isString(data)
-			? fs.write(fd, data, 0, 'utf8')
-			: fs
-					.write(fd, data, 0, data.length)
-					.then(() => fs.fsync(fd))
-					.then(() => fs.close(fd));
-	});
-}
-
-export function writeFileAtomic(
-	path: string,
+): Promise<void> {
+	const file = await fs.open(_path, 'w');
+	if (typeof data === 'string') {
+		await file.write(data, 0, 'utf8');
+	} else {
+		await file.write(data, 0, data.length);
+	}
+	await file.sync();
+	await file.close();
+}
+
+export async function writeFileAtomic(
+	_path: string,
 	data: string | Buffer,
-): Bluebird<void> {
-	return Bluebird.resolve(writeAndSyncFile(`${path}.new`, data)).then(() =>
-		fs.rename(`${path}.new`, path),
-	);
+): Promise<void> {
+	await writeAndSyncFile(`${_path}.new`, data);
+	await fs.rename(`${_path}.new`, _path);
+}
+
+export async function safeRename(src: string, dest: string): Promise<void> {
+	await fs.rename(src, dest);
+	const file = await fs.open(path.dirname(dest), 'r');
+	await file.sync();
+	await file.close();
 }
 
-export function safeRename(src: string, dest: string): Bluebird<void> {
-	return Bluebird.resolve(fs.rename(src, dest))
-		.then(() => fs.open(Path.dirname(dest), 'r'))
-		.tap(fs.fsync)
-		.then(fs.close);
+export async function existsAsync(_path: string): Promise<boolean> {
+	try {
+		await fs.access(_path);
```

Comment:
```suggestion
export async function existsAsync(path: string): Promise<boolean> {
	try {
		await fs.access(path);
```

Reply:
Results in `no-shadowed-variable  Shadowed name: 'path'`. I'll rename this param to `pathname` to be more explicit, since `_path` or `Path` are semantically not as correct.